### ISSUE TITLE
cannot import Hasql on windows

### ISSUE THREAD

**Issue by KeesBleijenberg** [nikita-volkov/hasql#50]
After upgrading to GHC version 8.0.1. I finally could install Hasql on win7 64 bits.
cabal install hasql => no errors. If I do cabal install again, it reports hasql is already installed.

Then I tried my first program:

module Main
where
import qualified Hasql as H

main :: IO ()
main = putStrLn "hello"
I did ghd --make testHasql.hs => Failed to load interface for `Hasql`. Use -v to see....
I did ghc --make -v testHasql.hs and got the error message below.

I wonder whether Hasql needs SSL at runtime or can I connect to postgresql without SSL. I got a popup during installation complaining about a missing SSL dll. 
But the installation of Hasql didn't abort.

What can I do?
Kees

Glasgow Haskell Compiler, Version 8.0.1, stage 2 booted by GHC version 7.10.2
Using binary package database: D:\Program Files\Haskell Platform\8.0.1\lib\package.conf.d\package.cache
Using binary package database: C:\Users\kees\AppData\Roaming\ghc\x86_64-mingw32-8.0.1\package.conf.d\package.cache
loading package database D:\Program Files\Haskell Platform\8.0.1\lib\package.conf.d
loading package database C:\Users\kees\AppData\Roaming\ghc\x86_64-mingw32-8.0.1\package.conf.d
wired-in package ghc-prim mapped to ghc-prim-0.5.0.0
wired-in package integer-gmp mapped to integer-gmp-1.0.0.1
wired-in package base mapped to base-4.9.0.0
wired-in package rts mapped to rts
wired-in package template-haskell mapped to template-haskell-2.11.0.0
wired-in package ghc mapped to ghc-8.0.1
wired-in package dph-seq not found.
wired-in package dph-par not found.
Hsc static flags: 
loading package database D:\Program Files\Haskell Platform\8.0.1\lib\package.conf.d
loading package database C:\Users\kees\AppData\Roaming\ghc\x86_64-mingw32-8.0.1\package.conf.d
wired-in package ghc-prim mapped to ghc-prim-0.5.0.0
wired-in package integer-gmp mapped to integer-gmp-1.0.0.1
wired-in package base mapped to base-4.9.0.0
wired-in package rts mapped to rts-1.0
wired-in package template-haskell mapped to template-haskell-2.11.0.0
wired-in package ghc mapped to ghc-8.0.1
wired-in package dph-seq not found.
wired-in package dph-par not found.
*** Chasing dependencies:
Chasing modules from: *testHaSql.hs
!!! Chasing dependencies: finished in 0.00 milliseconds, allocated 0.322 megabytes
Stable obj: []
Stable BCO: []
Ready for upsweep
  [NONREC
      ModSummary {
         ms_hs_date = 2016-06-28 11:51:08.0848841 UTC
         ms_mod = Main,
         ms_textual_imps = [(Nothing, Prelude), (Nothing, Hasql)]
         ms_srcimps = []
      }]
*** Deleting temp files:
Deleting: 
compile: input file testHaSql.hs
*** Checking old interface for Main:
[1 of 1] Compiling Main             ( testHaSql.hs, testHaSql.o )
*** Parser [Main]:
!!! Parser [Main]: finished in 0.00 milliseconds, allocated 0.125 megabytes
*** Renamer/typechecker [Main]:
!!! Renamer/typechecker [Main]: finished in 15.60 milliseconds, allocated 5.820 megabytes

testHaSql.hs:4:1: error:
    Failed to load interface for `Hasql'
    Locations searched:
      Hasql.hs
      Hasql.lhs
      Hasql.hsig
      Hasql.lhsig
Upsweep partially successful.
*** Deleting temp files:
Deleting: 
link(batch): upsweep (partially) failed OR
   Main.main not exported; not linking.
*** Deleting temp files:
Deleting: 
*** Deleting temp dirs:
Deleting:

**Comment by nikita-volkov** [comment]
It seems like you're merely missing the library. Might be there's something wrong with your package database. I don't know, but it seems like the issue is on your end.

**Comment by KeesBleijenberg** [comment]
Thanks for your help

totally removed ghc from my computer (deleted all directory's for packages and run the deinstaller)
installed haskell platform 8.0.1 again
cabal update
cabal install postgresql-libpq
cabal install hasql 
During the installation of hasql itself I get a popup window saying: The program can't start because SSLAY32.dll is missing from your computer. This dll has something to do with SSL.
But hasql is installed (no error message in the terminal)
If I compiled the test program I get the same error as described earlier. The error message is not very helpfull.
Is this a SSL problem? I should be able to connect to postgresql without SSL or...? So the SSLAY32.dll isn't needed (no dynamic loading). Anyway, I'll try to fix this dll error first.

**Comment by KeesBleijenberg** [comment]
Extra info: even if sslay32.dll is on the path and on my computer, hasql installer still complains about missing sslay32.dll. What triggers this search for sslay32.dll?

**Comment by nikita-volkov** [comment]
Hasql doesn't deal with SSL at all. Can be some of the transitive dependencies.

**Comment by KeesBleijenberg** [comment]
Hasql is the last thing that gets installed. Only during that(!) install the error message pops up. 
Any idea why my testprogram that does nothing then importing hasql, fails to compile?
How can I troubleshoot this problem?

**Comment by ruslantalpa** [comment]
@KeesBleijenberg how about removing everything haskell related (cabal/ghc) and try using `stack`
http://docs.haskellstack.org/en/stable/install_and_upgrade/#windows

**Comment by nikita-volkov** [comment]
@KeesBleijenberg In your private email you've said that you've solved the problem. So what was it?

**Comment by KeesBleijenberg** [comment]
Hi,

 

English is not my native language. So maybe I didn’t tell it the right way….

 

I removed GHC from my system and installed stack (GHC 7.10.3, 64 bits Windows). Then I added HaSQL to the cabal file of my test program with only a import of HaSQL in it. Stack installed HaSQL. This time without problems. No pop ups warning for missing dll ‘s. The test program  now also compiles without errors.

 

Kees

 

 

 

Van: Nikita Volkov [mailto:notifications@github.com] 
Verzonden: maandag 4 juli 2016 11:27
Aan: nikita-volkov/hasql
CC: KeesBleijenberg; Mention
Onderwerp: Re: [nikita-volkov/hasql] cannot import Hasql on windows (#50)

 

@KeesBleijenberg <https://github.com/KeesBleijenberg>  In your private email you've said that you've solved the problem. So what was it?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nikita-volkov/hasql/issues/50#issuecomment-230246644> , or mute the thread <https://github.com/notifications/unsubscribe/AKyhLZzzGqfZ8BCskxyRze4WCGKP_sRzks5qSNHygaJpZM4JAB9d> .Afbeelding verwijderd door afzender.

Geen virus gevonden in dit bericht.
Gecontroleerd door AVG - www.avg.com
Versie: 2016.0.7640 / Virusdatabase: 4613/12553 - datum van uitgifte: 07/04/16
